pkg/proxy/backend/backend_conn_mgr.go

```diff
@@ -282,20 +281,6 @@ func (mgr *BackendConnManager) tryRedirect(ctx context.Context) {
 	mgr.backendConn = newConn
 }
 
-// The original db in the auth info may be dropped during the session, so we need to authenticate with the current db.
-// The user may be renamed during the session, but the session cannot detect it, so this will affect the user.
-// TODO: this may be a security problem: a different new user may just be renamed to this user name.
-func (mgr *BackendConnManager) updateAuthInfoFromSessionStates(sessionStates []byte) error {
```

Comment:
It's a mistake that it's not used. It should be called, or the default `db` should be set to "" in the second handshake.

Reply:
Sure. Please revert this function.

Reply:
Done, /cc @djshow832